modules/c/examples/code_snippets/main.c

```diff
@@ -1078,3 +1197,392 @@ Placeholder for Date accessors.
+    bool docs_example_resetRequired = argResetRequired;
+    // tag::p2p-act-rep-start-full[]
+
+    CBLReplicator* thisRepl =
+        CBLReplicator_Create(&argConfig, &err); // <.>
+
+    // end::p2p-act-rep-start-full[]
+    if(!docs_example_resetRequired) {
+    // tag::p2p-act-rep-start-full[]
+
+      CBLReplicator_Start(thisRepl,false); // <.>
+
+    // end::p2p-act-rep-start-full[]
+    } else {
+    // tag::replication-reset-checkpoint[]
+      CBLReplicator_Start(thisRepl, true); // <.>
+
+    // end::replication-reset-checkpoint[]
+    }
+
+    return thisRepl;
+
+}
+
+// END replication.html >> initialize section
+
+// PAGE=Data Sync >> Monitor
+// URL=https://docs.couchbase.com/couchbase-lite/current/c/replication.html#lbl-repl-mon
+// BEGIN replication.html >> Monitor section
+//
+static void docs_act_replication_Monitor(
+                                       void* context,
+                                       CBLReplicator* argRepl) {
+
+    CBLError err;
+
+    CBLReplicator* thisRepl = argRepl;
+
+    // tag::p2p-act-rep-add-change-listener[]
+    // Purpose -- illustrate addition of a Replicator change listener
+    CBLListenerToken* token_ReplChangeListener =
+            CBLReplicator_AddChangeListener(thisRepl,
+                                            simpleChangeListener,
+                                            NULL);
+
+    // end::p2p-act-rep-add-change-listener[]
+    // tag::add-document-replication-listener[]
+    // Purpose -- illustrate addition of a Document Replicator  listener
+    CBLListenerToken* token_ReplDocListener =
+            CBLReplicator_AddDocumentReplicationListener(
+                                                        thisRepl,
+                                                        SimpleReplicationDocumentListener,
+                                                        context);
+
+    // end::add-document-replication-listener[]
+    // tag::remove-document-replication-listener[]
+    // Purpose -- illustrate removal of a listener
+
+    CBLListener_Remove(token_ReplDocListener);
+
+    CBLListener_Remove(token_ReplChangeListener);
+
+    // end::remove-document-replication-listener[]
+
+    // tag::p2p-act-rep-status[]
+    // Purpose -- illustrate use of CBLReplicator_Status()
+
+    CBLReplicatorStatus thisState = CBLReplicator_Status(thisRepl);
+
+    if(thisState.activity==kCBLReplicatorStopped) {
+        if(thisState.error.code==0) {
+            CBLReplicator_Start(thisRepl,false);
+        } else {
+            printf("Replicator stopped -- code %d", thisState.error.code);
+            // ... handle error ...
+        }
+    }
+
+    // end::p2p-act-rep-status[]
+
+
+    // tag::replication-pendingdocuments[]
+
+    FLDict thisPendingIdList =
+        CBLReplicator_PendingDocumentIDs(thisRepl, &err); // <.>
+
+    if(!FLDict_IsEmpty(thisPendingIdList)) {
+        FLDictIterator item;
+        FLDictIterator_Begin(thisPendingIdList, &item);
+        FLValue itemValue;
```

Comment:
This value is never initialized with anything and will surely cause a crash if you tried to run this snippet.  I think you are missing something that is assigning it.

Reply:
Right. Will fix.